lib/signs/utilities/audio.ex

```diff
@@ -137,70 +50,71 @@ defmodule Signs.Utilities.Audio do
     end
   end
 
-  defp get_prediction_announcements({audios, sign}, items) do
+  defp get_prediction_announcements({audios, sign}, messages) do
     {new_audios, sign} =
-      Stream.filter(items, &match?({:predictions, _}, &1))
-      |> Enum.flat_map_reduce(sign, fn {:predictions, messages}, sign ->
-        Stream.with_index(messages)
-        |> Enum.flat_map_reduce(sign, fn {message, index}, sign ->
+      Stream.filter(messages, &match?(%Message.Predictions{}, &1))
+      |> Enum.flat_map_reduce(sign, fn message, sign ->
+        Stream.with_index(message.predictions)
+        |> Enum.flat_map_reduce(sign, fn {prediction, index}, sign ->
+          {minutes, _} = PaEss.Utilities.prediction_minutes(prediction, message.terminal?)
+
           cond do
             # Announce boarding if configured to. Also, if we normally announce arrivals, but the
             # prediction went straight to boarding, announce boarding instead.
-            match?(%Message.Predictions{minutes: :boarding}, message) &&
-              message.prediction.trip_id not in sign.announced_boardings &&
-                (announce_boarding?(sign, message) ||
-                   (announce_arriving?(sign, message) &&
-                      message.prediction.trip_id not in sign.announced_arrivals)) ->
-              {Audio.TrainIsBoarding.from_message(message),
-               update_in(sign.announced_boardings, &cache_value(&1, message.prediction.trip_id))}
+            minutes == :boarding &&
+              prediction.trip_id not in sign.announced_boardings &&
+                (announce_boarding?(sign, prediction) ||
+                   (announce_arriving?(sign, prediction) &&
+                      prediction.trip_id not in sign.announced_arrivals)) ->
```

Comment:
Would it be worth taking the opportunity to separate out some of these long conditions into helpers for readability's sake?

Reply:
That's a good idea. Or something like pulling out each announcement clause, since they're basically independent. I'll experiment with that a bit in a separate PR.